packages/fxa-settings/src/models/integrations/web-integration.ts

```diff
@@ -0,0 +1,29 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import {
+  BaseIntegration,
+  IntegrationFeatures,
+  IntegrationType,
+} from './base-integration';
+
+export class WebIntegration extends BaseIntegration {
+  constructor() {
+    super(IntegrationType.Web);
+  }
+
+  get features(): IntegrationFeatures {
+    return {
+      ...super.features,
+      reuseExistingSession: true,
+      fxaStatus: this.isFxaStatusSupported(),
+    };
+  }
+
+  private isFxaStatusSupported(): boolean {
+    // TODO: check if `navigator.userAgent` is firefox desktop.
```

Comment:
I can do this TODO in this PR as well, though, I need to double check where this is actually used heh.